test/test_loader.py

```diff
@@ -0,0 +1,477 @@
+    root = tmpdir.mkdir('loader_mods')
+    mod_file = root.join('file_mod.py')
+    mod_file.write(MOCK_MODULE_CONTENT)
+
+    test_mod, _ = loader.load_module(
+        'file_mod', mod_file.strpath, imp.PY_SOURCE)
+    callables, jobs, shutdowns, urls = loader.clean_module(
+        test_mod, tmpconfig)
+
+    assert len(callables) == 2
+    assert test_mod.first_command in callables
+    assert test_mod.second_command in callables
+    assert len(jobs) == 2
+    assert test_mod.interval5s in jobs
+    assert test_mod.interval10s in jobs
+    assert len(shutdowns)
+    assert test_mod.shutdown in shutdowns
+    assert len(urls) == 1
+    assert test_mod.example_url in urls
+
+    # ignored function is ignored
+    assert test_mod.ignored not in callables
+    assert test_mod.ignored not in jobs
+    assert test_mod.ignored not in shutdowns
+    assert test_mod.ignored not in urls
+
+
+def test_clean_callable_default(tmpconfig, func):
+    loader.clean_callable(func, tmpconfig)
+
+    # Default values
+    assert hasattr(func, 'unblockable')
+    assert func.unblockable is False
+    assert hasattr(func, 'priority')
+    assert func.priority == 'medium'
+    assert hasattr(func, 'thread')
+    assert func.thread is True
+    assert hasattr(func, 'rate')
+    assert func.rate == 0
+    assert hasattr(func, 'channel_rate')
+    assert func.rate == 0
+    assert hasattr(func, 'global_rate')
+    assert func.global_rate == 0
+    assert hasattr(func, 'event')
+    assert func.event == ['PRIVMSG']
+
+    # Not added by default
+    assert not hasattr(func, 'rule')
+    assert not hasattr(func, 'commands')
+    assert not hasattr(func, 'intents')
+
+
+def test_clean_callable_event(tmpconfig, func):
+    setattr(func, 'event', ['low', 'UP', 'MiXeD'])
+    loader.clean_callable(func, tmpconfig)
+
+    assert hasattr(func, 'event')
+    assert func.event == ['LOW', 'UP', 'MIXED']
+
+
+def test_clean_callable_event_string(tmpconfig, func):
+    setattr(func, 'event', 'some')
+    loader.clean_callable(func, tmpconfig)
+
+    assert hasattr(func, 'event')
+    assert func.event == ['SOME']
+
+
+def test_clean_callable_rule(tmpconfig, func):
+    setattr(func, 'rule', [r'abc'])
+    loader.clean_callable(func, tmpconfig)
+
+    assert hasattr(func, 'rule')
+    assert len(func.rule) == 1
+
+    # Test the regex is compiled properly
+    regex = func.rule[0]
+    assert regex.match('abc')
+    assert regex.match('abcd')
+    assert not regex.match('efg')
+
+
+def test_clean_callable_rule_string(tmpconfig, func):
+    setattr(func, 'rule', r'abc')
+    loader.clean_callable(func, tmpconfig)
+
+    assert hasattr(func, 'rule')
+    assert len(func.rule) == 1
+
+    # Test the regex is compiled properly
+    regex = func.rule[0]
+    assert regex.match('abc')
+    assert regex.match('abcd')
+    assert not regex.match('efg')
+
+
+def test_clean_callable_rule_nick(tmpconfig, func):
+    """Assert ``$nick`` in a rule will match ``Sopel: `` or ``Sopel, ``."""
```

Comment:
Probably worth defining the bot's nick in `tmpconfig`, just in case the default value changes at some point. The tests don't need to start failing because of that.

Same for the `_nickname` test below, and others that assume this default.